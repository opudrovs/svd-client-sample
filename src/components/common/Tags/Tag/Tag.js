import PropTypes from 'prop-types';

import styles from './Tag.module.scss';

/**
 * Tag.
 */

const Tag = ({ text }) => (
    <div className={styles.tag}>{text}</div>
);

Tag.propTypes = {
    text: PropTypes.string.isRequired
};

export default Tag;
