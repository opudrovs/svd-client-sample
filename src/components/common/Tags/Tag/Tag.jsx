import { memo } from 'react';

import PropTypes from 'prop-types';

import styles from './Tag.module.scss';

/**
 * Tag component.
 */
const Tag = ({ text }) => <div className={styles.tag}>{text}</div>;

Tag.propTypes = {
  text: PropTypes.string.isRequired,
};

export default memo(Tag);
