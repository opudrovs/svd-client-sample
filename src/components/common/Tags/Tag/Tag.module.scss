.tag {
  overflow: hidden;
  max-width: 100%;
  @include remValues('padding', 4px, 8px, 4px, 8px);
  border: 1px solid $brandingColor13;
  @include rem('border-radius', 4px);
  @include rem('margin-bottom', 4px);
  @include rem('margin-left', 4px);
  color: #333;
  @include tagFont;
  text-align: center;
  white-space: nowrap;
  text-overflow: ellipsis;
}
