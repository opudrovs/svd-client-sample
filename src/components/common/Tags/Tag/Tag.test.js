/* REACT */

import React from 'react';

/* LIBRARIES */

import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

/* COMPONENTS */

import Tag from './Tag.js';

const text = 'Test text';


/* TESTS */

describe('With Enzyme', () => {
    it(`Component shows '${text}'`, () => {
        const component = shallow(<Tag text={text} />);

        expect(component.find('div')
            .text())
            .toEqual(text);
    });
});

/* SNAPSHOT TESTS */

describe('With Snapshot Testing', () => {
    it(`Component shows '${text}'`, () => {
        const component = renderer.create(<Tag text={text} />);
        const tree = component.toJSON();

        expect(tree)
            .toMatchSnapshot();
    });
});