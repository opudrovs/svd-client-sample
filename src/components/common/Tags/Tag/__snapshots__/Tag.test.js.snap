// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing Component shows 'Test text' 1`] = `
<div
  className="tag"
>
  Test text
</div>
`;
